src/state/medicaltypes/reducer.ts

```diff
@@ -0,0 +1,115 @@
+import produce from "immer";
+import { IAction } from "../types";
+import {
+  CREATE_MEDICALTYPE_FAIL,
+  CREATE_MEDICALTYPE_LOADING,
+  CREATE_MEDICALTYPE_RESET,
+  CREATE_MEDICALTYPE_SUCCESS,
+  GET_MEDICALTYPE_SUCCESS,
+  GET_MEDICALTYPE_LOADING,
+  GET_MEDICALTYPE_FAIL,
+  EDIT_MEDICALTYPE_FAIL,
+  EDIT_MEDICALTYPE_LOADING,
+  EDIT_MEDICALTYPE_RESET,
+  EDIT_MEDICALTYPE_SUCCESS,
+  DELETE_MEDICALTYPE_LOADING,
+  DELETE_MEDICALTYPE_SUCCESS,
+  DELETE_MEDICALTYPE_FAIL,
+} from "./consts";
+import { initial } from "./initial";
+import { IMedicalTypesState } from "./types";
+
+export default produce((draft: IMedicalTypesState, action: IAction<any, any>) => {
+  switch (action.type) {
+    /**
```

Comment:
Please I suggest to keep only declaration that are usefull now? What do you think?